docs/API_POLICY.md

```diff
@@ -0,0 +1,64 @@
+# Hub API Policy
+
+This document proposes a policy regarding making updates to the APIs of this repo.
+
+NOTE: Hub Release Version and API version are different and may not be in sync. Making a Hub major release does not necessarily increment API version too. API version will be incremented only when there is a breaking change.
+eg. `/v1` -> `/v2` only when there is a breaking change.
+
+Hub API service hosts two types of APIs:
+
+## 1. Versioned APIs
+
+- These are the APIs which are exposed for external users to use and integrate with their applications.
+- These will have support provided by the Hub team.
+- These are the APIs which return hub resource data.
+- The version of the APIs will be prefixed to the API route. 
+  Eg. resource API - /v1/resources
+- Swagger Documentation for these APIs can be found at `https://api.hub.tekton.dev/<version>/schema/swagger.json`. Replace `<version>` with current version of APIs.
+ 
+
+#### Additive changes
+
+- Additive changes are changes that are added to the APIs and do not cause problems for existing users of the API.
+- If the current version of API is `/v1` then any new addition for eg. adding new field in response of a API will not affect existing users of v1 version of that API.
+- These changes must be approved by at least 2 OWNERS.
+
+
+#### Breaking changes
+
+- Breaking changes are those when users have to change their implementation as per the changes in API. for eg. removal of a field from response, removing an API, removing a query param of an API. 
+- These changes will be introduced as a new version of the APIs.
+- If current version of APIs is `/v1` and an API is going to be removed then all other APIs will be moved to `/v2`. 
+  Users of v1 will be able to use old versioned APIs for 9 months then they will be removed.
```

Comment:
Are we saying that one a `v2` of an API is out, 9 months after, `v1` gets removed ?
(note : we should probably use "at least 9 month" 😝 )

Reply:
yup once `v2` is out `v1` will get removed after 9 months.

Reply:
but let's say there is some breaking change in one of the API in `v1` so increment the version to `v2` so all other `v1` APIs would also be incremented or just the one?

Reply:
my understanding was we increment all the APIs to v2 and then after 9 months remove all the v1 APIs

Reply:
yeah, it is probably better to bump the API version for all at the same time.